security-whitepaper/cloud-native-security-whitepaper.md

```diff
@@ -0,0 +1,846 @@
+- Configuration signing - signing of a config file, i.e. application config files: most common in the case of a GitOps approach, where there can be a process to validate and check configurations.
+- Package signing - Signing of a package of artifacts, like application packages.
+
+For generic software artifacts such as libraries or OCI artifacts, signing these artifacts indicates their provenance of approved usage by the organization. Verification of these artifacts is equally crucial in ensuring that only the authorized artifacts are allowed. It is strongly recommended that repositories require mutual authentication to make changes to images in registries or to commit code to repositories.
+
+##### Encryption
+
+Container Image Encryption encrypts a container image so that its contents are confidential. The container image contents are encrypted to ensure that they remain confidential for promotion from build time through runtime. In the event of a compromised distribution, the image's registry contents remain secret, which can help for use cases such as protecting trade secrets or other confidential material.
+
+Another common use of Container Image Encryption is to enforce container image authorization. When image encryption is coupled with key management attestation and/or authorization and credential distribution, it is possible to require that a container image can only run on particular platforms. Container image authorization is useful for compliance use cases such as geo-fencing or export control and digital rights media management.
+
+### Deploy
+
+![Figure 4](RackMultipart20201111_figure4.png)
+
+_Figure 4_
+
+The "Deploy" phase is responsible for incorporating a sequence of 'pre-flight' checks in order to ensure that the applications that are going to be deployed in the runtime environment conform and comply with organization wide security and compliance policies.
+
+#### Pre-Flight Deployment Checks
+
+Prior to deployment, organizations should verify the existence, applicability, and current state of:
+
+- Image signature and integrity
+- Image runtime policies (e.g absence of malware or critical vulnerabilities)
+- Container runtime policies (e.g absence of excessive privileges)
+- Host vulnerability and compliance controls
+- Workload, application, and network security policies
+
+#### Observability & Metrics
+
+Instituting observability and metrics into cloud native architectures delivers security insights so appropriate stakeholders can resolve and mitigate anomalies appearing in reporting; tools in this area can help collect and visualize this information. Through the use of behavioral and heuristic analysis teams can detect and escalate outliers, suspicious events, and unexplained calls to appropriate stakeholders. Artificial intelligence (AI), [machine learning (ML)](#kix.tlhq6ssgript), or statistical modeling are all mechanisms that are encouraged to assist in behavioral and heuristic analysis development.
```

Comment:
Is the referenced link for ML intentional ?

Reply:
@PushkarJ it's not - will be fixed.